### ISSUE TITLE
Feature Request: Configure IoT CloudWatch Logging

### ISSUE THREAD

**Issue by rraub** [hashicorp/terraform-provider-aws#7620]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

I would like to be able to configure IoT region logging from Terraform, currently its only configurable via the aws web interface or the aws cli. The log configuration just takes a role and a log level to use for Cloudwatch. There can only be one IoT log setting per region.

### Potential Terraform Configuration

```hcl
resource "aws_iot_log" foobar" {
  role_arn: "foo"
  level: "info"
  region: "us-east-1"
}
```

### References

* https://docs.aws.amazon.com/cli/latest/reference/iot/set-logging-options.html

**Comment by bflad** [comment]
The [IOT API reference for `SetLoggingOptions`](https://docs.aws.amazon.com/iot/latest/apireference/API_SetLoggingOptions.html) recommends using `SetV2LoggingOptions` instead:

> NOTE: use of this command is not recommended. Use SetV2LoggingOptions instead.

A potential configuration that follows the [`SetV2LoggingOptions` API](https://docs.aws.amazon.com/iot/latest/apireference/API_SetV2LoggingOptions.html) naming seems to be:

```hcl
resource "aws_iot_logging_options" "example" {
  default_log_level = "DEBUG|INFO|ERROR|WARN|DISABLED"
  disable_all_logs = true|false
  role_arn = "${aws_iam_role.example.arn}"
}
```

**Comment by dedoussis** [comment]
There is no API for deleting logging options. Should the delete method of this resource just `return nil`? Also no need for any update method - `force new` will do.

**Comment by jamescrowley** [comment]
@dedoussis perhaps could just set the log level to 'DISABLED' again, given the resource can't be created/deleted as such?

**Comment by ilya-korotya** [comment]
@bflad , @rraub . I have been working with the implementation of `aws_iot_logging_options`. In documentation https://docs.aws.amazon.com/iot/latest/apireference/API_SetV2LoggingOptions.html `roleArn` and `defaultLogLevel` is optional fields. But when I try to create first time this resource without `arnRole` or `defaultLogLevel` fields. I get the error:
```
Must set a roleArn and a defaultLogLevel the first time setLoggingOptionsV2 is called.
``` 
How can I set a required flag for these fields but only for the first run?

**Comment by rraub** [comment]
I agree, these should probably be required in the terraform resource. It seems that the intent of the optional fields is just to make the API calls easier to work with if your only updating one of the parameters but not the other.

**Comment by ilya-korotya** [comment]
Hi @bflad. I wrote a draft version of working with `aws_iot_logging_options`. But when I run [the test](https://github.com/terraform-providers/terraform-provider-aws/pull/13392/files#diff-f06083865f6e4205883e16d84d573b18R18) I get the following error:
```
        Error: Provider produced inconsistent result after apply
        
        When applying changes to aws_iot_logging_options.test, provider "aws" produced
        an unexpected new value for was present, but now absent.
        
        This is a bug in the provider, which should be reported in the provider's own
        issue tracker.
```
Could you help me with this error(check this [PR](https://github.com/terraform-providers/terraform-provider-aws/pull/13392))? I don't have any idea of how this fix 😞

**Comment by ewbankkit** [comment]
Closed via #23440.

**Comment by akupila** [comment]
@ewbankkit I believe the issue you referred to here is not the same as this one. This issue refers to [Iot Core logging](https://docs.aws.amazon.com/iot/latest/developerguide/configure-logging.html), while #23440 is for a rule that sends incoming messages to Cloudwatch
